### PR TITLE
Added -o to Quail CLI to change the default output directory

### PR DESCRIPTION
**Pull request by WilcoFiers**

### CODE REVIEW

File: bin/quail [hunk 1]
```diff
@@ -1,10 +1,44 @@
...
+/**
+ *
+ */
+function validateOutputDirectory(val) {
+  var stat;
+  // If it is a directory, then use it.
+  try {
+    stat = fs.lstatSync(val);
+    if (stat.isDirectory()) {
+      debugger;
```
Comment (jessebeach) [inline bin/quail]: Stray debugger;

File: bin/quail [hunk 2]
```diff
@@ -1,10 +1,44 @@
...
+      return val;
+    }
+  }
+  catch (error) {
```
Comment (jessebeach) [inline bin/quail]: ```process.exit(1)```;

File: bin/quail [hunk 3]
```diff
@@ -1,10 +1,44 @@
...
+    // Nothing to do here.
+  }
+  // Otherwise see if it's relative to the project. Make an assumption that paths
+  // are relative to the ROOT of the project, hence the '..'.
+  try {
+    stat = fs.lstatSync(path.join(__dirname, '..', val));
+    if (stat.isDirectory()) {
+      debugger;
```
Comment (jessebeach) [inline bin/quail]: Stray debugger;

File: bin/quail [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
+      return path.join(__dirname, '..', val);
+    }
+  }
+  catch (error) {
+    // Nothing to do here.
```
Comment (jessebeach) [inline bin/quail]: Exit the process

File: src/js/scripts/phantom_evaluator.js [hunk 5]
```diff
@@ -134,7 +135,7 @@ var len = 0;
 var date = new Date();
 var timestamp = [
   date.getFullYear(),
-  ('0' + date.getMonth()).slice(-2),
+  ('0' + (date.getMonth() + 1)).slice(-2),
```
Comment (jessebeach) [inline src/js/scripts/phantom_evaluator.js]: Ha! Good catch. Need to update the code in the master branch with this, too

Comment (jessebeach) [inline src/js/scripts/phantom_evaluator.js]: Looks like you already did! 307065ceb6ce71f9e2e50f6360b15162fa4a00dd

**Comment by jessebeach** [comment]
We're not going to merge this into master (definitely not dev, that's a defunct branch now). You already fixed the month bug in 307065ceb6ce71f9e2e50f6360b15162fa4a00dd. So, what I recommend is creating a branch from 2.2.15 tag (just make sure the name of the branch isn't simply 2.2.16. Name it something like 2.2.16-output-config). We won't do a pull request for this. We'll just cut a tag from the branch. For review, we can compare your work to the SHA 590733055f2f79cf94d958407ce9f14daa2e7e3a

**Comment by jessebeach** [comment]
Oh, and I should mention, it looks great. Just a few stray debuggers and opportunities to exit the process in a catch.

**Comment by WilcoFiers** [comment]
I'm a little confused about your comments for the branches? From what I can gather dev is what 2.2.x is developed from, and the work you're doing on master will (I assume) become part of 2.3.0 (or maybe 3.0.0?) once ready. Shouldn't we then keep using dev for patches on 2.2? Lets discuss in the call Monday. Not sure where you want the try/catch statements - as this is mostly just a copy of what you've made in master. Can you specify?

**Comment by jessebeach** [comment]
Redid the PR as https://github.com/quailjs/quail/pull/316

**Pull request closed without merging**